nib_ls test failure on SPARC

@yarikoptic - any thoughts on this one?

http://nipy.bic.berkeley.edu/builders/nibabel-py2.7-wheezy-sparc/builds/115

```
======================================================================
FAIL: nibabel.tests.test_scripts.test_nib_ls_multiple
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/buildslave/nd-bb-slave-sparc64-wheezy/nibabel-py2_7-wheezy-sparc/build/venv/local/lib/python2.7/site-packages/nose/case.py", line 197, in runTest
    self.test(*self.arg)
  File "/home/buildslave/nd-bb-slave-sparc64-wheezy/nibabel-py2_7-wheezy-sparc/build/venv/local/lib/python2.7/site-packages/nibabel/tests/test_scripts.py", line 94, in test_nib_ls_multiple
    "Full output was: %s" % stdout_lines)
AssertionError: Type sub-string didn't start with 'i'. Full output was: [u'/home/buildslave/nd-bb-slave-sparc64-wheezy/nibabel-py2_7-wheezy-sparc/build/venv/local/lib/python2.7/site-packages/nibabel/tests/data/example4d.nii.gz      <i2 [128,  96,  24,   2] 2.00x2.00x2.20x2000.00  #exts: 2 sform', u'/home/buildslave/nd-bb-slave-sparc64-wheezy/nibabel-py2_7-wheezy-sparc/build/venv/local/lib/python2.7/site-packages/nibabel/tests/data/example_nifti2.nii.gz <i2 [ 32,  20,  12,   2] 2.00x2.00x2.20x2000.00  #exts: 2 sform', u'/home/buildslave/nd-bb-slave-sparc64-wheezy/nibabel-py2_7-wheezy-sparc/build/venv/local/lib/python2.7/site-packages/nibabel/tests/data/small.mnc             <i2 [ 18,  28,  29]      9.00x8.00x7.00', u'/home/buildslave/nd-bb-slave-sparc64-wheezy/nibabel-py2_7-wheezy-sparc/build/venv/local/lib/python2.7/site-packages/nibabel/tests/data/nifti2.hdr            <i2 [ 91, 109,  91]      2.00x2.00x2.00']
```

--- a/nibabel/tests/test_scripts.py
+++ b/nibabel/tests/test_scripts.py
@@ -6,6 +6,7 @@ Test running scripts
 """
 from __future__ import division, print_function, absolute_import
 
+import sys
 import os
 from os.path import (dirname, join as pjoin, abspath, splitext, basename,
                      exists)
@@ -97,7 +98,9 @@ def test_nib_ls_multiple():
 
     # they should be indented correctly.  Since all files are int type -
     ln = max(len(f) for f in fnames)
-    assert_equal([l[ln:ln+2] for l in stdout_lines], [' i']*4,
+    little_endian = sys.byteorder == 'little'
+    assert_equal([l[ln:ln+2+int(not little_endian)] for l in stdout_lines],
+                 [' i']*4 if little_endian else [' <i']*4,
                  msg="Type sub-string didn't start with 'i'. "
                      "Full output was: %s" % stdout_lines)
     # and if disregard type indicator which might vary

--- a/nibabel/tests/test_scripts.py
+++ b/nibabel/tests/test_scripts.py
@@ -98,11 +98,10 @@ def test_nib_ls_multiple():
 
     # they should be indented correctly.  Since all files are int type -
     ln = max(len(f) for f in fnames)
-    little_endian = sys.byteorder == 'little'
-    assert_equal([l[ln:ln+2+int(not little_endian)] for l in stdout_lines],
-                 [' i']*4 if little_endian else [' <i']*4,
-                 msg="Type sub-string didn't start with 'i'. "
-                     "Full output was: %s" % stdout_lines)
+    i_str = ' i' if sys.byteorder == 'little' else ' <i'
+    assert_equal([l[ln:ln + len(i_str)] for l in stdout_lines], [i_str] * 4,
+          msg="Type sub-string didn't start with '%s'. "
+              "Full output was: %s" % (i_str, stdout_lines))
     # and if disregard type indicator which might vary
     assert_equal(
         [l[l.index('['):] for l in stdout_lines],

